src/server/db/prisma-client/survey.ts

```diff
@@ -196,9 +218,8 @@ export class SurveyPrismaClient {
             const questionsCreate = questions.map((question) => ({
                 questionText: question.questionText,
                 roles: {
-                    // Link the question to the existing roles
                     connect: question.roles.map((role) => ({
-                        id: role.id,
+                        id: allRoles.find((r) => r.role === role.role)?.id,
```

Comment:
Better to use `!` here instead of `?`, as it should always exist (and crash if it doesn't)